core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/enclave_plan_capabilities/enclave_plan_capabilities.go

```diff
@@ -0,0 +1,60 @@
+package enclave_plan_capabilities
+
+import (
+	"encoding/json"
+	"github.com/kurtosis-tech/kurtosis/container-engine-lib/lib/backend_interface/objects/service"
+	"github.com/kurtosis-tech/stacktrace"
+)
+
+// Use NewEnclavePlanCapabilitiesBuilder to construct this object
+type EnclavePlanCapabilities struct {
+	// we do this way in order to have exported fields which can be marshalled
+	// and an unexported type for encapsulation
```

Comment:
This isn't clear to me at all to be honest. I trust you this is needed, but I need more info to really understand why we need such a complex structure here...
If this is just to be able to have public attributes, honestly I think we should just put the attribute in this object even though they are public. This nested private struct is a lot more difficult to understand IMO rather than bearing the cost of having public attributes